pgpy/pgp.py

```diff
@@ -1767,7 +1767,10 @@ def _sign(self, subject, sig, **prefs):
                 uid = next(iter(self.parent.userids), None)
 
         if sig.hash_algorithm is None:
-            sig._signature.halg = next(h for h in uid.selfsig.hashprefs if h.is_supported)
+            if any(map(lambda h: h.is_supported, uid.selfsig.hashprefs)):
```

Comment:
`next()` has an optional second parameter to provide a default to return in the event that there is no next item to return, so we can rewrite this to not need to iterate twice:

```python
sig._signature.halg = next((h for h in uid.selfsig.hashprefs if .is_supported), HashAlgorithm.SHA256)
```

Reply:
nice, thanks!  TIL about the 2nd arg to `next()`…